### DISCUSSION TITLE
Log lines for an active step are inaccessible

### DISCUSSION THREAD

**Discussion by mcarans** [community/community#44250] [Actions]
### Select Topic Area

Bug

### Body

Describe the bug
When opening the "run" page for an active job you can see:

Gray checkmarks and expandable sections for completed steps and all of their log lines.
Yellow loading animation on any active steps.
Circles for the future steps.
The log lines for the active steps will then start streaming in, but the prior lines for that step are not there, and there appears to be no way to see them until the step completes.

To Reproduce

Run an action
Visit the run page after a few seconds and try to find the log lines for the currently running step that occurred before you loaded the page.

Expected behavior
All of the log lines that were emit before I loaded the page should be visible, then new ones stream in afterwards.

This bug has been raised twice and closed without resolution in the repo: https://github.com/actions/runner/issues/886 https://github.com/actions/runner/issues/2131. [Stefan Ruvceski](https://github.com/ruvceskistefan) (presumably a GitHub staff member) said to raise it here.

This issue has been a pain for me for a while. Whenever I have a step that runs far longer than expected, I cannot look at the log to see where it has reached. I can't determine whether it's near the finish line or stuck somewhere near the beginning.

If developer inconvenience is insufficient cause to fix this, then another way to look at it is cost. If I know a step is stuck near the beginning from the log, I will probably stop it rather than letting it run in the hope it is near completion. Imagine this unnecessary processing across many errant steps in multiple Actions from many projects and the resources add up.

This really needs to be prioritised and fixed.

**Comment by etaormm** [comment]
Bumping this topic. 
Hopefully someone will pick this up...

**Comment by ajschmidt8** [comment]
+1

This is a huge oversight on the part of the GitHub Actions team.

This is pretty basic CI functionality that is missing.

This should 100% be prioritized.

**Comment by milieu** [comment]
Response from Github engineering last month here:
https://github.com/actions/runner/issues/886#issuecomment-1405845909

**Comment by axel-h** [comment]
Related discussion: #35363

**Comment by mcarans** [comment]
Potentially related documentation issue: https://github.com/github/docs/issues/25041

**Comment by cmahnke** [comment]
Before reopening the linked bugs and discussions, was this really fixed?
Today I have noticed the same problem and was wondering it the has been a unwanted regression...

**Comment by eliotmiranda** [comment]
The is awful. I have workflows that take nearly an hour (builds on ARM32). The last thing I want to do is restart the workflow. But I need to see the workflow to know if a build has succeeded before the build is tested. Please, please prioritize and fix this bug!

**Comment by user202729** [comment]
Refer to https://github.com/actions/runner/issues/2131#issuecomment-2087346728 , only last 1000 lines are accessible, the following lines are streamed using some websocket, previous lines are lost. Although these appears to be only accessible through the web UI.

This is partially accessible through CLI as follows

1. go to the web UI (**requires login**) and get the job internal id. Search for string `data-job-step-backscroll-url`, and see something like the following

	```
	data-job-step-backscroll-url="/OWNER/REPO/actions/runs/11111111111/jobs/33333333333/steps/44444444-4444-4444-4444-444444444444/backscroll"
	```

2. the step id is already there, but if you want to get the current in-progress step do

	```bash
	run_id=11111111111
	job_internal_id=33333333333
	in_progress_step=$(curl "https://github.com/OWNER/REPO/actions/runs/$run_id/jobs/$job_internal_id/steps" \
		-H 'accept: application/json' | jq -r '.[] | select(.status == "in_progress") | .id')
	```

3. fetch the backscroll

	```bash
	curl "https://github.com/OWNER/REPO/actions/runs/$run_id/jobs/$job_internal_id/steps/$in_progress_step/backscroll" \
		-H 'accept: application/json' | jq -r '.lines.[].line'
	```

any idea how to automate step 1? My current method is

```
curl "https://github.com/OWNER/REPO/actions/runs/$run_id/job/$job_external_id" \
	-H "cookie: user_session=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" | grep backscroll
```

but the cookie need to be copied from the browser.

tl;dr full command

```
user_session="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
url='...'
run_id=$(cut -d/ -f8 <<< "$url")
owner=$(cut -d/ -f4 <<< "$url")
repo=$(cut -d/ -f5 <<< "$url")
job_external_id=$(echo $url | awk -F'[/?]' '{print $10}')
job_internal_id=$(curl "https://github.com/$owner/$repo/actions/runs/$run_id/job/$job_external_id" \
    -H "cookie: user_session=$user_session" | grep backscroll|head -n1|cut -d/ -f8)
in_progress_step=$(curl "https://github.com/$owner/$repo/actions/runs/$run_id/jobs/$job_internal_id/steps" \
	-H 'accept: application/json' | jq -r '.[] | select(.status == "in_progress") | .id')
curl "https://github.com/$owner/$repo/actions/runs/$run_id/jobs/$job_internal_id/steps/$in_progress_step/backscroll" \
	-H 'accept: application/json' | jq -r '.lines.[].line'
```

**Comment by ulzha** [comment]
Indeed, what's up with unability to tail logs? It's such a basic necessity. Especially when a job is doing lots and we suspect it's stuck, we want to inspect its logs logged so far.

The above sorts of workarounds shouldn't be necessary. (I don't seem to make them work either. `curl -L --fail-with-body "https://github.my-corp.com/$owner/$repo/actions/runs/$run_id/jobs/$job_id/steps -H "cookie: user_session=***; __Host-user_session_same_site=***" -H 'accept: application/json'` returns HTTP 406.)
